Updates basic tutorials for Cadence 1.0 and fixes old migration commands

joshuahannan opened the pull request:
Closes https://github.com/onflow/docs/issues/769

I went through and made sure all the code looked good for the first 6 tutorials and that important new concepts like entitlements were explained properly.

I had to rush through the Fungible Token tutorial a bit since USDC stuff is the priority now, so it is a little sloppy, but at least it has updated code now.

The playground links are not updated yet because I'm waiting for either the Cadence 1.0 playground to become the flagship playground or when I finish updating the other tutorials, whichever comes first. Right now, the links still use old playground and old code, but the code examples in the tutorials themselves have Cadence 1.0 code to show how it works.

I also updated some of the commands in the migration guide to be correct

gregsantos commented on versioned_docs/version-1.0/cadence-migration-guide/index.md:
```diff
@@ -40,6 +40,8 @@ Help is available during the [Cadence 1.0 Office Hours](https://calendar.google.
 #### Expectations
 
 - Any contracts that are not upgraded will fail after Testnet and Mainnet upgrade.
+  They **WILL NOT** be able to be upgraded again, so it is imperative
```
```suggestion
  They **cannot** be upgraded again, so it is imperative
```

gregsantos commented on versioned_docs/version-1.0/tutorial/01-first-steps.md:
```diff
@@ -15,7 +15,7 @@ Cadence introduces new features to smart contract programming that help develope
   by ensuring that resources (and their associated assets) can only exist in one location at a time, cannot be copied, and cannot be accidentally lost or deleted
 - Built-in pre-conditions and post-conditions for functions and [transactions](../language/transactions.md)
-- The utilization of capability-based security, which enforces access control by requiring that access to objects
+- The utilization of capability-based security and entitlements, which enforces access control by requiring that access to objects
```
```suggestion
- The utilization of capability-based security and entitlements, which enforces access control by requiring that access to objects,
```

joshuahannan replied:
no comma needed here. We're saying that the access is restricted, not that capability security is restricted

gregsantos commented on versioned_docs/version-1.0/tutorial/01-first-steps.md:
```diff
@@ -24,14 +24,19 @@ Please see the [Cadence introduction](../index.md) for more information about th
 The Flow Playground should be compatible with any standard web browser,
 but we recommend that you use Google Chrome with it,
-because it has been tested and optimized for only the Chrome browser so far.
+because it has been tested and optimized primarily with the Chrome browser so far.
```
The Flow Playground should work with any standard web browser. However, we recommend using Google Chrome, as it has been primarily tested and optimized for this browser.

gregsantos commented on versioned_docs/version-1.0/tutorial/01-first-steps.md:
```diff
@@ -24,14 +24,19 @@ Please see the [Cadence introduction](../index.md) for more information about th
+because it has been tested and optimized primarily with the Chrome browser so far.
+
+The [Flow Crescendo (Cadence 1.0) upgrade](https://flow.com/upgrade/crescendo)
+introduces new powerful features to Cadence. Currently, the playground uses an older version
```
```suggestion
introduces new, powerful features to Cadence. Currently, the playground uses an older version
```

gregsantos commented on versioned_docs/version-1.0/tutorial/01-first-steps.md:
```diff
@@ -40,20 +45,70 @@ with deploying Cadence smart contracts and interacting with transaction and scri
+Normally, the terms of a contract are supervised and enforced by a trusted third party, such as a bank or a lawyer.
+
+A smart contract is a computer program stored in a network like a blockchain
+that verifies and executes the performance of a contract (like a lawyer does)
```
```suggestion
that verifies and executes the performance of a contract
```

joshuahannan replied:
I disagree with this one. I like the clarification.

gregsantos commented on versioned_docs/version-1.0/tutorial/01-first-steps.md:
```diff
@@ -40,20 +45,70 @@ with deploying Cadence smart contracts and interacting with transaction and scri
+that verifies and executes the performance of a contract (like a lawyer does)
+without the need for any trusted third party anywhere in the process, because the code itself is trusted.
+
+Programs that run on blockchains are commonly referred to as smart contracts
+because they mediate important functionality (such as currency)
+without having to rely on a central authority (like a bank).
```
```suggestion
Programs that run on blockchains are commonly referred to as smart contracts because they facilitate important functions, such as managing digital currency, without relying on a central authority like a bank.
```

gregsantos reviewed (approved):
few copy nits but lg2G

sisyphusSmiling commented on versioned_docs/version-1.0/tutorial/03-resources.md:
```diff
@@ -529,26 +521,25 @@ Optionals allow developers to account for `nil` cases more gracefully.
+We use the nil-coalescing operator (`??`) to "unwrap" the optional.
+This basically means that we are handling the case where the `load` method returns `nil`.
+If it returns `nil`. We `panic`, which will abort execution of the transaction
+with an error message. It is important for developers to always to provide detailed error messages
```
```suggestion
with an error message. It is important for developers to always provide detailed error messages
```

sisyphusSmiling commented on versioned_docs/version-1.0/tutorial/05-non-fungible-tokens-2.md:
```diff
@@ -371,32 +358,30 @@ Now, back to the tutorial!
 
 When a user stores their NFT `Collection` in their account storage, it is by default not available for other users to access.
-A user's authorized account object (`AuthAccount`, which gives access to private storage)
+A user's authorized account object (`&Account`, which gives access to private storage)
```
> authorized account object

Maybe "entitled account"? Only suggesting because earlier it's mentioned that `&Account` is publicly accessible and grants access to /public/ namespace.

joshuahannan replied:
That is a good point. I'll reword it to make it more clear

sisyphusSmiling reviewed (approved):
Looks great!

joshuahannan commented:
I addressed all the comments. Doesn't seem like there is any blockers here, so I'm just going to merge. 

I created two more issues to track the rest of the work required for the tutorials and assigned them to myself, so I'll be able to get to those after the USDC work is done:
* https://github.com/onflow/docs/issues/800
* https://github.com/onflow/docs/issues/799
